pkg/ccl/sqlproxyccl/proxy_handler.go

```diff
@@ -539,6 +545,36 @@ func (handler *proxyHandler) handle(ctx context.Context, incomingConn net.Conn)
 	}
 }
 
+// validateRequest validates the incoming connection by ensuring that the SQL
+// connection knows some additional information about the tenant (i.e. the
+// cluster name) before being allowed to connect.
+func (handler *proxyHandler) validateConnection(
+	ctx context.Context, tenantID roachpb.TenantID, clusterName string,
+) error {
+	tenant, err := handler.directoryCache.LookupTenant(ctx, tenantID)
+	if err != nil {
```

Comment:
nit: consider this alternative implementation of validateConnection. The current behavior is subtle and violates a few common conventions around how go error handling is done.

```go
tenant, err := handler.directoryCache.LookupTenant(ctx, tenantID)
if err != nil && status.Code(err) != codes.NotFound {
    return err
}
if err == nil && tenant.ClusterName == clusterName {
    return nil
}
return err = withCode(
	errors.Newf("cluster %s-%d not found", clusterName, tenantID.ToUint64()),
	codeParamsRoutingFailed,
)
```

Reply:
Done.